libp2p/protocols/kademlia/kademlia.nim

```diff
@@ -316,15 +316,16 @@ proc putValue*(
 
   let peers = await kad.findNode(key)
 
-  # TODO: only query max of alpha peers (selectAlphaPeers)
-  let rpcBatch = peers.mapIt(kad.switch.dispatchPutVal(it, key, value))
-
   kad.dataTable.insert(key, value, $times.now().utc)
-  try:
-    await rpcBatch.allFutures().wait(kad.config.timeout)
-  except AsyncTimeoutError:
-    # Dispatch will timeout if any of the calls don't receive a response (which is normal)
-    discard
+
+  var chunk: seq[PeerId]
+  forChunks(peers, kad.config.alpha, chunk):
+    let rpcBatch = chunk.mapIt(kad.switch.dispatchPutVal(it, key, value))
```

Comment:
subject to same comment as here: https://github.com/vacp2p/nim-libp2p/pull/1764/files#r2433013327